src/Stratis.Bitcoin/Connection/PeerLifetimeBehavior.cs

```diff
@@ -0,0 +1,97 @@
+using Microsoft.Extensions.Logging;
+using NBitcoin.Protocol;
+using NBitcoin.Protocol.Behaviors;
+using Stratis.Bitcoin.Base;
+
+namespace Stratis.Bitcoin.Connection
+{
+    /// <summary>
+    /// A behaviour that will manage the lifetime of peers.
+    /// </summary>
+    public class PeerLifetimeBehavior : NodeBehavior
+    {
+        /// <summary>Logger factory to create loggers.</summary>
+        private readonly ILoggerFactory loggerFactory;
+
+        /// <summary>Handle the lifetime of a peer.</summary>
+        private readonly IPeerLifetime peerLifetime;
+
+        /// <summary>Instance logger.</summary>
+        private readonly ILogger logger;
+
+        /// <summary>Instance of the <see cref="ConnectionManager"/>.</summary>
+        public ConnectionManager ConnectionManager { get; private set; }
+
+        /// <summary>Instance of the <see cref="ChainHeadersBehavior"/> that belongs to the same peer as this behaviour.</summary>
+        private ChainHeadersBehavior chainHeadersBehavior;
+
+        /// <summary>Instance of the <see cref="ConnectionManagerBehavior"/> that belongs to the same peer as this behaviour.</summary>
+        private ConnectionManagerBehavior connectionManagerBehavior;
+
+        public PeerLifetimeBehavior(IConnectionManager connectionManager, ILoggerFactory loggerFactory, IPeerLifetime peerLifetime)
+        {
+            this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
+            this.loggerFactory = loggerFactory;
+            this.peerLifetime = peerLifetime;
+            this.ConnectionManager = connectionManager as ConnectionManager;
```

Comment:
this is offtopic question, but what is the purpose of this? don't we want to work with the interface?

Reply:
This is now removed. but yes you are right, however its a different PR I would say.